fontforge/plugin.c

```diff
@@ -0,0 +1,558 @@
+/* Copyright (C) 2000-2012 by George Williams, 2021 Skef Iterum */
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+
+ * The name of the author may not be used to endorse or promote products
+ * derived from this software without specific prior written permission.
+
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#include <fontforge-config.h>
+
+#ifndef _NO_PYTHON
+
+#include "plugin.h"
+
+#include "gfile.h"
+#include "uiinterface.h"
+
+int use_plugins = true; // Prefs variable
+int attempted_plugin_load = false;
+
+enum plugin_startup_mode_type plugin_startup_mode = sm_ask;
+GList_Glib *plugin_data = NULL;
+
+void FreePluginEntry(PluginEntry *pe) {
+    free(pe->name);
+    free(pe->package_name);
+    free(pe->summary);
+    free(pe->package_url);
+    free(pe->module_name);
+    free(pe->attrs);
+    if (pe->pyobj != NULL) {
+        Py_DECREF(pe->pyobj);
+    }
+    if (pe->entrypoint != NULL) {
+        Py_DECREF(pe->entrypoint);
+    }
+    free(pe);
+}
+
+static PluginEntry *NewPluginEntry(char *name, char *pkgname, char *modname, char *url,
+                                   enum plugin_startup_mode_type sm) {
+    PluginEntry *pe = malloc(sizeof(PluginEntry));
+    pe->name = name;
+    pe->package_name = pkgname;
+    pe->summary = NULL;
+    pe->module_name = modname;
+    pe->attrs = NULL;
+    pe->package_url = url;
+    pe->startup_mode = sm;
+    pe->pyobj = NULL;
+    pe->entrypoint = NULL;
+    pe->is_present = false;
+    pe->is_well_formed = true;
+    pe->has_prefs = false;
+    return pe;
+}
+
+static char *getPluginDirName() {
```

Comment:
Total honesty: I just copy styles from various parts of the code because I don't know what to do otherwise and wind up with a hodgepodge because the existing code is a hodgepodge. Tell me what to do with things like function and variable names and I'm not likely to put up much resistance (unlike some other areas, I guess).

Reply:
My preference would be to at least keep the casing consistent within a file, especially new files; with pascal case for functions unless replacing stuff that's already following some other pattern.